src/types.cpp

```diff
@@ -57,8 +59,31 @@ std::ostream &operator<<(std::ostream &os, const SizedType &type)
   return os;
 }
 
+bool SizedType::IsSameType(const SizedType &t) const
```

Comment:
It isn't really obvious to me the difference between `IsSameType` and `IsEqual`. Is the only difference that type size isn't checked? Maybe `IsEqualIgnoringSize` then?